zallet/src/components/json_rpc/payments.rs

```diff
@@ -0,0 +1,446 @@
+                _ => other,
+            },
+            PrivacyPolicy::AllowRevealedSenders => match other {
+                PrivacyPolicy::FullPrivacy | PrivacyPolicy::AllowRevealedAmounts => self,
+                PrivacyPolicy::AllowRevealedRecipients => PrivacyPolicy::AllowFullyTransparent,
+                _ => other,
+            },
+            PrivacyPolicy::AllowFullyTransparent => match other {
+                PrivacyPolicy::FullPrivacy
+                | PrivacyPolicy::AllowRevealedAmounts
+                | PrivacyPolicy::AllowRevealedRecipients
+                | PrivacyPolicy::AllowRevealedSenders => self,
+                PrivacyPolicy::AllowLinkingAccountAddresses => PrivacyPolicy::NoPrivacy,
+                _ => other,
+            },
+            PrivacyPolicy::AllowLinkingAccountAddresses => match other {
+                PrivacyPolicy::FullPrivacy
+                | PrivacyPolicy::AllowRevealedAmounts
+                | PrivacyPolicy::AllowRevealedSenders => self,
+                PrivacyPolicy::AllowRevealedRecipients | PrivacyPolicy::AllowFullyTransparent => {
+                    PrivacyPolicy::NoPrivacy
+                }
+                _ => other,
+            },
+            PrivacyPolicy::NoPrivacy => self,
+        }
+    }
+
+    /// This policy is compatible with a given policy if it is identical to or less strict
+    /// than the given policy.
+    ///
+    /// For example, if a transaction requires a policy no stricter than
+    /// [`PrivacyPolicy::AllowRevealedSenders`], then that transaction can safely be
+    /// constructed if the user specifies [`PrivacyPolicy::AllowLinkingAccountAddresses`],
+    /// because `AllowLinkingAccountAddresses` is compatible with `AllowRevealedSenders`
+    /// (the transaction will not link addresses anyway). However, if the transaction
+    /// required [`PrivacyPolicy::AllowRevealedRecipients`], it could not be constructed,
+    /// because `AllowLinkingAccountAddresses` is _not_ compatible with
+    /// `AllowRevealedRecipients` (the transaction reveals recipients, which is not
+    /// allowed by `AllowLinkingAccountAddresses`.
+    pub(super) fn is_compatible_with(&self, other: Self) -> bool {
+        self == &self.meet(other)
+    }
+
+    pub(super) fn allow_revealed_amounts(&self) -> bool {
+        self.is_compatible_with(PrivacyPolicy::AllowRevealedAmounts)
+    }
+
+    pub(super) fn allow_revealed_recipients(&self) -> bool {
+        self.is_compatible_with(PrivacyPolicy::AllowRevealedRecipients)
+    }
+
+    pub(super) fn allow_revealed_senders(&self) -> bool {
+        self.is_compatible_with(PrivacyPolicy::AllowRevealedSenders)
+    }
+
+    pub(super) fn allow_fully_transparent(&self) -> bool {
+        self.is_compatible_with(PrivacyPolicy::AllowFullyTransparent)
+    }
+
+    pub(super) fn allow_linking_account_addresses(&self) -> bool {
+        self.is_compatible_with(PrivacyPolicy::AllowLinkingAccountAddresses)
+    }
```

Comment:
These helpers are present in `zcashd` but are currently unused here. We need to figure out what `zcashd` used them for and perform equivalent checks here, or remove them if truly unnecessary.

Reply:
Now handled.